test/batch-submitter/transaction-batch-submitter.spec.ts

```diff
@@ -282,3 +297,74 @@ describe('TransactionBatchSubmitter', () => {
     })
   })
 })
+
+describe('TransactionBatchSubmitter to Ganache', () => {
+  let signer
+  const server = ganache.server({
+    default_balance_ether: 420,
+    blockTime: 2_000
+  })
+  const provider = new Web3Provider(ganache.provider())
+
+  before(async () => {
+    await server.listen(3001)
+    signer = await provider.getSigner()
+  })
+
+  after(async () => {
+    await server.close()
+  })
+
+  // Unit test for getReceiptWithResubmission function,
+  // tests for increasing gas price on resubmission
+  it('should resubmit a transaction if it is not confirmed', async () => {
+    const gasPrices = []
+    const numConfirmations = 2
+    const sendTxFunc = async (gasPrice) => {
+      // push the retried gasPrice
+      gasPrices.push(gasPrice)
+
+      const tx = signer.sendTransaction({
+        to: DECOMPRESSION_ADDRESS,
+        value: 88,
+        nonce: 0,
+        gasPrice
+      })
+
+      const response = await tx
+
+      return signer.provider.waitForTransaction(
+        response.hash,
+        numConfirmations,
+        100_000
```

Comment:
Shoot! Sorry @annieke I just realized that it might actually be bad for us to include a timeout in the `waitForTransaction` calls! Eek! 

Here's the edge case I didn't consider:

1. Batch submitter submits a bunch of transactions. The `maxGasPrice` is too low. Transactions get stuck in the mempool.
2. The timeout is reached, causing the batch submitter to attempt to resubmit.
3. **By default** Ethers will use the latest nonce in the **pending** transactions not just the last **confirmed** transaction. This means that we might end up submitting the same batch at nonce `x` and nonce `x+1`! Aka we'd end up submitting the same batch twice 😭

Reply:
just removed this and the config vars around it!